airbyte-webapp/src/views/Connection/ConnectionForm/components/ScheduleField.tsx

```diff
@@ -0,0 +1,182 @@
+import { Field, FieldInputProps, FieldProps } from "formik";
+import { ChangeEvent, useMemo, useState } from "react";
+import { useIntl } from "react-intl";
+
+import { ControlLabels, DropDown, DropDownRow, Input } from "components";
+
+import {
+  ConnectionScheduleData,
+  ConnectionScheduleDataBasicSchedule,
+  ConnectionScheduleDataCron,
+  ConnectionScheduleType,
+} from "core/request/AirbyteClient";
+
+import availableCronTimeZones from "../../../../config/availableCronTimeZones.json";
+import { ConnectionFormMode } from "../ConnectionForm";
+import { useFrequencyDropdownData } from "../formConfig";
+import styles from "./ScheduleField.module.scss";
+
+interface ScheduleFieldProps {
+  scheduleData: ConnectionScheduleData | undefined;
+  scheduleType: ConnectionScheduleType | undefined;
+  mode: ConnectionFormMode;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
+  onDropDownSelect?: (item: DropDownRow.IDataItem) => void;
+}
+
+const ScheduleField: React.FC<ScheduleFieldProps> = ({
+  scheduleData,
+  scheduleType,
+  mode,
+  setFieldValue,
+  onDropDownSelect,
+}) => {
+  const [isCustomSchedule, setIsCustomSchedule] = useState<boolean>(scheduleType === ConnectionScheduleType.cron);
+  const { formatMessage } = useIntl();
+  const frequencies = useFrequencyDropdownData(scheduleData);
+
+  const onScheduleChange = (
```

Comment:
When testing this PR locally, I found a bug where if I switch from Manual to some basic frequency (e.g. Every hour), then it says `form is invalid` until I click somewhere. This should probably only really be investigated after refactoring this component as I suggested above, since that could potentially fix the bug.

Reply:
Another bug I found -- when I choose a cron string and click Save, the Replication Frequency dropdown switches back to the previously selected value (e.g. Manual) while the loading spinner is spinning, then switches back to Cron once the save is complete. Again, probably only worth it to investigate this after refactoring this component.

Reply:
I didn't see this happening after the remake

Reply:
@harshithmullapudi I'm no longer seeing the second bug, but I'm still seeing the one that I mentioned in my first comment here, where it says the form is invalid when I switch from manual to something else, until I click. I have taken a screen recording to demonstrate this:

I'm not sure why this is happening but it may have something to do with how the Formik field values are being set when switching from one to the other. Could you investigate this issue more?

https://user-images.githubusercontent.com/22731524/189013108-17f4ea1d-1aef-49d8-968a-20c8a004834f.mov

Reply:
Yeah I have resolved it. Also updated the video in the PR description